### PR TITLE
Add PyTorch operator-level SDC detection module

### PR DESCRIPTION
**Pull request by hechin**
Adds `pytorch_op/` module for operator-level GPU SDC (Silent Data Corruption) detection.

  Unlike model-level benchmarks that run full training workloads, this module isolates
  individual PyTorch operators and compares results across GPUs to catch
  deterministic-but-wrong errors at the operator level.
  
  ### Modes
  - **self** — Self-consistency: runs each op N times on a single GPU, checks bit-exact
  reproducibility
  - **cross-gpu** — Cross-GPU comparison: runs the same op on all GPUs with identical
  inputs, compares results (recommended)
  - **sweep** — Systematic HBM coverage: staircase memory fill to test different GPU memory
  regions

  ### Presets
  - `quick` — 1 iteration, all ops, 4kx4k (~1-2 min)
  - `standard` — 3 iterations, all ops + pipelines (~5 min)
  - `thorough` — 10 iterations + full HBM sweep (~25-30 min)

  ### Coverage
  - 85 individual ops: arithmetic, matrix, reduction, activation, convolution, memory,
  normalization, attention, embedding, pooling, scatter/gather, sorting, backward pass
  - 15 composite pipelines: BERT/GPT-2/LLaMA transformer layers, SwiGLU MLP, training steps
  with optimizer
  - Precision modes: fp32, tf32, fp16, bf16, fp8
  - Supports NVIDIA and AMD GPUs

  ## Usage
  
  ```bash
  python -m pytorch_op --preset quick               # Quick SDC check, all GPUs
  python -m pytorch_op --preset quick --fail-fast    # Stop on first failure
  python -m pytorch_op --preset quick -v             # Full detail on failures
  python -m pytorch_op --mode sweep                  # Systematic full-HBM scan
  python -m pytorch_op --mode self                   # Self-test all GPUs
  python -m pytorch_op --list-ops                    # List all registered ops

  Test plan

  Tested on 8x AMD MI300X (Instinct) host:

  $ python -m pytorch_op --preset quick --duration-limit 60
  ALL CROSS-GPU TESTS PASSED - 85 comparisons across 8 GPUs in 57.3s
  Result: {'total_comparisons': 85, 'passed': 85, 'failed': 0}

### CODE REVIEW

**Review by canbyjiao — APPROVED** [review]
Looks good to me!
